### DISCUSSION TITLE
How to embed generated subtitles with timing? Like each spoken word is highlighted - any open source for this?

### DISCUSSION THREAD

**Discussion by FurkanGozukara** [openai/whisper#1957] [Q&A]
What i mean is like in this video each word is highlighted and also embedded to the video

Do you know how can we do such thing? 

https://twitter.com/GozukaraFurkan/status/1745466705592164835

Lets say I have generated subtitles with accurate time via Whisper

How can we embed them to the video?

**Comment by ryanheise** [comment]
For inspiration, take a look at https://github.com/openai/whisper/discussions/435#discussioncomment-4238872 which makes a video out of audio, but you may be able to adapt it to suit your purposes. That snippet was using a .ass file as input, but Whisper can now generate an SRT file with word highlighting like this:

```bash
whisper --output_format srt --word_timestamps True --highlight_words True YOURINPUTFILE
```

Once you have that SRT file, you can use a command like the one in the linked comment to try to overlay that subtitles file over a video, but use your SRT file instead of a .ass file.

**Reply by FurkanGozukara** [reply]
awesome looking right now

**Reply by FurkanGozukara** [reply]
i made .ass file like this but it is not working any ideas? i get no error. 

I added .txt to upload here.

[a.ass.txt](https://github.com/openai/whisper/files/13931860/a.ass.txt)

**Reply by ryanheise** [reply]
I'm saying you should use the SRT file instead of .ass because Whisper doesn't generate .ass files, it only generates SRT files. Here:

> but use your SRT file instead of a .ass file.

And to clarify, by that, I mean use the SRT file generated using the command I gave you.

**Reply by FurkanGozukara** [reply]
ok i made it work :)

**Comment by FurkanGozukara** [comment]
> For inspiration, take a look at [#435 (reply in thread)](https://github.com/openai/whisper/discussions/435#discussioncomment-4238872) which makes a video out of audio, but you may be able to adapt it to suit your purposes. That snippet was using a .ass file as input, but Whisper can now generate an SRT file with word highlighting like this:
> 
> ```shell
> whisper --output_format srt --word_timestamps True --highlight_words True YOURINPUTFILE
> ```
> 
> Once you have that SRT file, you can use a command like the one in the linked comment to try to overlay that subtitles file over a video, but use your SRT file instead of a .ass file.
